Cannot build the manager after pulling branch 4.9.0

|Wazuh version|Component|Install type|
|---|---|---|
| 4.9.0 | Build |Manager |

1. Having Wazuh manager built on an old 4.9.0 branch.
2. Pull branch 4.9.0:
```shell
git pull
```
4. Clean build and dependencies:
```shell
make clean-deps
make clean
```
5. Build the manager:
```shell
make TARGET=server DEBUG=1
```

I always get a bunch of errors like this:

```
In file included from /root/wazuh-4.9.0/src/shared_modules/content_manager/src/components/factoryDownloader.hpp:16,
                 from /root/wazuh-4.9.0/src/shared_modules/content_manager/src/components/factoryContentUpdater.hpp:18,
                 from /root/wazuh-4.9.0/src/shared_modules/content_manager/src/actionOrchestrator.hpp:16,
                 from /root/wazuh-4.9.0/src/shared_modules/content_manager/src/onDemandManager.hpp:20,
                 from /root/wazuh-4.9.0/src/shared_modules/content_manager/src/onDemandManager.cpp:12:
/root/wazuh-4.9.0/src/shared_modules/content_manager/src/components/APIDownloader.hpp: In member function ‘void APIDownloader::downloadContent()’:
/root/wazuh-4.9.0/src/shared_modules/content_manager/src/components/APIDownloader.hpp:84:30: error: no matching function for call to ‘IURLRequest::download(HttpURL, std::string&, const APIDownloader::downloadContent()::<lambda(const string&, long int)>&, <brace-enclosed initializer list>, <brace-enclosed initializer list>, std::string&)’
   84 |         m_urlRequest.download(
      |         ~~~~~~~~~~~~~~~~~~~~~^
   85 |             HttpURL(m_url), m_fullFilePath, onError, {}, {}, m_context->spUpdaterBaseContext->httpUserAgent);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

## Cause

`make clean-deps` is not cleaning _shared/http-request_.

## Proposed fix

Let Makefile's `clean` target erase folder _shared/http-request_ content.

--- a/src/Makefile
+++ b/src/Makefile
@@ -2727,7 +2727,7 @@ ifneq ($(wildcard external/cpython/*),)
 endif
 
 clean-deps:
-	rm -rf $(EXTERNAL_DIR) $(EXTERNAL_CPYTHON) external/$(WPYTHON_TAR) $(EXTERNAL_ROCKSDB)
+	rm -rf $(EXTERNAL_DIR) $(EXTERNAL_TAR) shared_modules/http-request/* shared_modules/http-request/.??*
 
 clean-internals: clean-unit-tests
 	rm -f $(BUILD_SERVER)

